### PR TITLE
P&G Dishwasher Challenge

### PR DESCRIPTION
**Pull request by kyordhel**
The *Set the Table and Clean Up* test was simplified focusing mostly in the cleaning part and following Procter & Gamble requirements. The robot no longer serves any meal (so far untried), and focuses in manipulation of cutlery, tableware, and spot removal.

Changes:

- Test renamed to *Procter & Gamble Dishwasher Challenge*
    - Internal files renamed
- Setup:
    - Test takes place in the arena (kitchen)
    - Table is already set, robot cleans
    - Dishwasher is open
    - Dishwasher has all racks out
    - There is only a spot (high contrast)
- Task:
    - Standard test start: door
    - All cutlery and tableware is moved to the dishwasher
        - A tray can be used instead of placing in the dishwasher racks
        - The tray should be placed in the dishwasher afterewards
    - Placement of _Tide**POD**_ in dishwasher
    - Remove a spot from the table using sponge/cloth
    - Robot shall leave (bonus points for leaving)
- Additional rules:
    - All teams can participate regardless of stage
- Scoresheet
    - Reworked
    - Simplified scoring
- Added *Procter & Gamble Award*

### CODE REVIEW

File: tests/PnG.tex [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+\subsection{Focus}
+This test focuses on object perception, manipulation, and planning.
+
+\subsection{Setup}
+\begin{itemize}
+	\item \textbf{Location:} This test takes place in the arena. A dining table is located close to the dishwasher.
+	\item \textbf{Dishwasher:} The dishwasher is near the table, preferably located in the same room. The dish washer is open and with all racks out.
+	\item \textbf{Tray:} A plastic tray is located either on top of the dishwasher, or onto one of its racks. The tray may have tableware and cutlery placed inside already.
+	\item \textbf{Table setting:} The table has several objects disposed in a typical setting for a meal for one person. These objects include tableware (e.g. place mats, napkins, dishes, glasses), and silverware (e.g. forks, spoons, knives).
+	\item \textbf{Spot:} There is a spot on the table next to the table setting.
```
Comment (balkce) [inline tests/PnG.tex]: Change this to: "There is a dirty spot on the table next to the table setting that requires cleaning."

File: tests/PnG.tex [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+\end{itemize}
+
+\subsection{Task}
+	\begin{enumerate}
+		\item \textbf{Entering the arena:} The robot enters the arena and navigates to the designated location.
+		\item \textbf{Fetching command:} The operator requests the robot to clean up the table.
```
Comment (balkce) [inline tests/PnG.tex]: Do we really need a speech command here?

File: tests/PnG.tex [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+		\item \textbf{Clean the table:} The robot takes all the tableware and cutlery to either the dishwasher or to the tray, as instructed (team's choice).
+		\item \textbf{Filling the dishwasher:} If the robot placed the objects into the tray, it must proceed to put the tray onto one of the dishwasher racks.
+		\item \textbf{Place the \TidePod:} The robot places the  \TidePod~into the dishwasher, preferrably inside the tab compartment.
```
Comment (balkce) [inline tests/PnG.tex]: Tidepod -> Cascade Pod

File: tests/PnG.tex [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+\subsection{Additional rules and remarks}
+\begin{enumerate}
+	\item \textbf{Collisions:} Slightly touching the table is allowed, as well as slightly pushing some objects. However, driving over the objects or any other form of a major collision is not allowed, and the referees directly stop the robot (Section \refsec{rule:safetyfirst}).
+
+	\item \textbf{Objects:} A total of 6 objects are used in this test following the distribution shown below:
+	\begin{itemize}
+		\item\textit{Silverware}: Any two objects.
+		\item\textit{Tableware}: Any three objects, excluding silverware. At least one must be a dish.
+		\item\textit{\TidePod}: One \TidePod.
```
Comment (balkce) [inline tests/PnG.tex]: Tidepod -> Cascade Pod

File: tests/PnG.tex [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+	\end{itemize}
+	All objects used in this test are taken from the list of standard objects (See \ref{rule:scenario_objects}). All of them are considered to be known to the robot.
+
+	\item \textbf{Safe placing:} Objects placed in the rack or tray must be placed gently ans safely. It must be clear to the referee that the robot is trying to put the object in place and not dumping, throwing, or dropping it. Dumped and dropped objects won't be scored even if they land in the dishwasher/tray.
```
Comment (balkce) [inline tests/PnG.tex]: typo: ans -> and

File: tests/PnG.tex [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+	\item \textbf{Special Objects:} The plastic tray and the \TidePod are provided by \textit{Procter \& Gamble} and are considered special objects. Teams are not allowed to use substitutes of them in this test.
```
Comment (balkce) [inline tests/PnG.tex]: Tidepod -> Cascade Pod

File: tests/PnG.tex [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+	However, scoring in this test will be only considered for those teams who have advanced to Stage II. This way, no Stage I team can have an overall score higher than a Stage II team.
+
+\end{enumerate}
+
+\subsection{Data recording}
+Please record the following data (See \refsec{rule:datarecording}):
+\begin{itemize}
+	\item Images of recognized objects
+	\item List of moved items
+\end{itemize}
```
Comment (balkce) [inline tests/PnG.tex]: As mentioned in the today's skype meeting, no data collecting this year.

File: tests/PnG.tex [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+\subsection{OC instructions}
+During Setup days:
+\begin{itemize}
+	\item Provide official cutlery and tableware for training.
+\end{itemize}
+
+2 hours before the test:
+\begin{itemize}
+	\item Announce the predefined location to take the command.
+	\item Announce the predefined location of the \TidePod.
```
Comment (balkce) [inline tests/PnG.tex]: Tidepod -> Cascade Pod

File: tests/PnG.tex (hunk 3 above)
Comment (kyordhel) [reply tests/PnG.tex]: Why Cascade Pod? I understood TidePOD was P&G family

image

**Comment by kyordhel** [comment]
@Loy, shouldn't be much of a problem. What about TidePod challenge?

File: tests/PnG.tex (hunk 3 above)
Comment (balkce) [reply tests/PnG.tex]: Fabio stated so in his email.

Comment (kyordhel) [reply tests/PnG.tex]: Fair enough!

File: tests/PnG.tex (hunk 2 above)
Comment (kyordhel) [reply tests/PnG.tex]: To make it look more natural. I will comment the line though.

**Comment by balkce** [comment]
This is fine, however, there seems to be some conflicts with this merge. When they're resolved, I'll be glad to approve the merge.

**Comment by airglow** [comment]
I would optionally (like in Storing Groceries) add 50 points for opening the dishwasher. Ideally we get a dishwasher that is open-able by a robot i.e. with a handle.

**Comment by kyordhel** [comment]
@balkce the conflicts were solved, but before merging, please take a look to #404 to check @airglow's proposal

**Pull request merged**